tests/unit/test_compare.py

```diff
@@ -445,44 +490,54 @@ def test_metrics_show_mocked(mocker, markdown):
 
 def test_metrics_show_default(capsys):
     show_metrics(
-        {
-            "metrics.yaml": {
-                "x.b": {"old": 5, "new": 6},
-                "a.d.e": {"old": 3, "new": 4, "diff": 1},
-                "a.b.c": {"old": 1, "new": 2, "diff": 1},
-            }
-        }
+        metrics={
+            "branch_1": {
+                "data": {
+                    "metrics.json": {"data": {"b": {"ad": 1, "bc": 2}, "c": 4}}
+                }
+            },
+            "branch_2": {
+                "data": {
+                    "metrics.json": {"data": {"a": 1, "b": {"ad": 3, "bc": 4}}}
+                }
+            },
+        },
+        all_branches=True,
     )
     out, _ = capsys.readouterr()
     assert out == textwrap.dedent(
         """\
-        Path    diff    new    old
-        x.b     -       6      5
-        a.d.e   1       4      3
-        a.b.c   1       2      1
+        Revision    Path          a    b.ad    b.bc    c
+        branch_1    metrics.json  -    1       2       4
+        branch_2    metrics.json  1    3       4       -
         """
     )
 
 
 def test_metrics_show_md(capsys):
     show_metrics(
-        {
-            "metrics.yaml": {
-                "x.b": {"old": 5, "new": 6},
-                "a.d.e": {"old": 3, "new": 4, "diff": 1},
-                "a.b.c": {"old": 1, "new": 2, "diff": 1},
-            }
+        metrics={
+            "branch_1": {
+                "data": {
+                    "metrics.json": {"data": {"b": {"ad": 1, "bc": 2}, "c": 4}}
+                }
+            },
+            "branch_2": {
+                "data": {
+                    "metrics.json": {"data": {"a": 1, "b": {"ad": 3, "bc": 4}}}
+                }
+            },
         },
+        all_branches=True,
         markdown=True,
     )
     out, _ = capsys.readouterr()
     assert out == textwrap.dedent(
         """\
-        | Path   | diff   | new   | old   |
-        |--------|--------|-------|-------|
-        | x.b    | -      | 6     | 5     |
-        | a.d.e  | 1      | 4     | 3     |
-        | a.b.c  | 1      | 2     | 1     |
+        | Revision   | Path         | a   | b.ad   | b.bc   | c   |
+        |------------|--------------|-----|--------|--------|-----|
+        | branch_1   | metrics.json | -   | 1      | 2      | 4   |
+        | branch_2   | metrics.json | 1   | 3      | 4      | -   |
```

Comment:
Original PR:
https://github.com/iterative/dvc/pull/5161
The test is passing and from "behavioral" point of view is correct, it was checking how `diff` command results behave when put into `show_metrics`. Modified the test so that test data matches usual output fed to `show_metrics`.
I had to fix it here, since I am changing `metrics.show` results schema, and those tests would no longer be passing.